src/components/DownloadButtonAndMenu.tsx

```diff
@@ -7,8 +6,18 @@ import {
 } from "@/components/ui/dropdown-menu"
 import { Button } from "@/components/ui/button"
 import { Download, ChevronDown } from "lucide-react"
+import { useDownloadJSON } from "@/hooks/use-download-json"
 
 export function DownloadButtonAndMenu({ className }: { className?: string }) {
+  const createBlobURL = (content: string) => {
+    const blob = new Blob([content], { type: "text/plain" })
+    return URL.createObjectURL(blob)
+  }
+
+  const { jsonContent, jsonFileName } = useDownloadJSON()
```

Comment:
Bad hook name, this hook should be named to indicate its used to download the current circuit json

useDownloadCurrentSnippetCircuitJson()